src/protocols/fiat-protocol.js

```diff
@@ -0,0 +1,291 @@
+ * @property {string} cryptoAsset - The provider-specific code of the crypto asset to sell.
+ * @property {string} fiatCurrency - The currency's ISO 4217 code (e.g., 'USD').
+ * @property {string} [refundAddress] - The wallet address to receive refunds in case of failure. Defaults to the account's address.
+ */
+
+/**
+ * @typedef {Object} SellExactCryptoAmountOptions
+ * @property {number | bigint} cryptoAmount - The amount of crypto asset to sell, in its base unit (e.g., wei for ETH).
+ * @property {never} [fiatAmount] - The amount of fiat currency to receive, in its smallest unit (e.g., cents for USD).
+ */
+
+/**
+ * @typedef {Object} SellForFiatAmountOptions
+ * @property {number | bigint} fiatAmount - The amount of fiat currency to receive, in its smallest unit (e.g., cents for USD).
+ * @property {never} [cryptoAmount] - The amount of crypto asset to sell, in its base unit (e.g., wei for ETH).
+ */
+
+/**
+ * A protocol-agnostic, standardized object representing a quote for an on/off-ramp transaction.
+ * @typedef {Object} FiatQuote
+ * @property {bigint} cryptoAmount - The amount of the crypto asset, in its base unit (e.g., wei).
+ * @property {bigint} fiatAmount - The amount of the fiat currency, in its smallest unit (e.g., cents).
+ * @property {bigint} fee - The fee charged for the transaction, denominated in the smallest unit of the fiat currency.
+ * @property {string} rate - The effective exchange rate, expressed as a string to avoid precision loss (e.g., a rate of "3000.50" for ETH/USD means 1 ETH = 3000.50 USD). Note: This rate applies to the standard units (e.g., ETH and USD).
+ * @property {Record<string, unknown>} [metadata] - Provider-specific raw data for the quote.
+ */
+
+/**
+ * @interface
+ */
+export class IFiatProtocol {
+  /**
+   * Gets a quote for a crypto asset purchase.
+   * @param {Omit<BuyOptions, 'recipient'>} options - The options for the buy operation.
+   * @returns {Promise<FiatQuote>} A quote for the transaction.
+   */
+  async quoteBuy (options) {
+    throw new NotImplementedError('quoteBuy(options)')
+  }
+
+  /**
+   * Generates a widget URL for a user to purchase a crypto asset with fiat currency.
+   * @param {BuyOptions} options - The options for the buy operation.
+   * @returns {Promise<string>} The URL for the user to complete the purchase.
+   */
+  async buy (options) {
+    throw new NotImplementedError('buy(options)')
+  }
+
+  /**
+   * Gets a quote for a crypto asset sale.
+   * @param {Omit<SellOptions, 'refundAddress'>} options - The options for the sell operation.
+   * @returns {Promise<FiatQuote>} A quote for the transaction.
+   */
+  async quoteSell (options) {
+    throw new NotImplementedError('quoteSell(options)')
+  }
+
+  /**
+   * Generates a widget URL for a user to sell a crypto asset for fiat currency.
+   * @param {SellOptions} options - The options for the sell operation.
+   * @returns {Promise<string>} The URL for the user to complete the sale.
```

Comment:
Instead of returning the url as a raw string, we should encapsulate it in a 'SellResult' type:

```javascript
/**
 * @typedef {Object} SellResult
 * @property {string} url - The url for the user to complete the sale.
 */
```

```suggestion
   * @returns {Promise<SellResult>} The operation's result.
```